packages/features/settings/outOfOffice/OutOfOfficeEntriesList.tsx

```diff
@@ -1,194 +1,356 @@
-                  <div className="w-12" />
-                </div>
-                <div className="dark:bg-darkgray-50 absolute -top-3 left-3 -z-10 h-[70px] w-[70px] rotate-[24deg] rounded-3xl border-2 border-[#e5e7eb] p-8 opacity-60 dark:opacity-90">
-                  <div className="w-12" />
-                </div>
-                <div className="dark:bg-darkgray-50 relative z-0 flex h-[70px] w-[70px] items-center justify-center rounded-3xl border-2 border-[#e5e7eb] bg-white">
-                  <Icon name="clock" size={28} />
-                  <div className="dark:bg-darkgray-50 absolute right-4 top-5 h-[12px] w-[12px] rotate-[56deg] bg-white text-lg font-bold" />
-                  <span className="absolute right-4 top-3 font-sans text-sm font-extrabold">z</span>
-                </div>
+  const searchParams = useCompatSearchParams();
+  const selectedTab = searchParams?.get("type") ?? OutOfOfficeTab.MINE;
+
+  const { data, isPending, fetchNextPage, isFetching, refetch } =
+    trpc.viewer.outOfOfficeEntriesList.useInfiniteQuery(
+      {
+        limit: 10,
+        fetchTeamMembersEntries: selectedTab === OutOfOfficeTab.TEAM,
+        searchTerm: selectedTab === OutOfOfficeTab.TEAM ? debouncedSearchTerm : undefined,
+        recordType: recordType,
+      },
+      {
+        getNextPageParam: (lastPage) => lastPage.nextCursor,
+        placeholderData: keepPreviousData,
+      }
+    );
+
+  useEffect(() => {
+    refetch();
+    if (selectedTab === OutOfOfficeTab.MINE) {
+      setDebouncedSearchTerm("");
+    }
+  }, [oooEntriesAdded, oooEntriesUpdated, deletedEntry, selectedTab, refetch]);
+
+  const totalDBRowCount = data?.pages?.[0]?.meta?.totalRowCount ?? 0;
+  //Flatten the array of arrays from the useInfiniteQuery hook
+  const flatData = useMemo(
+    () => data?.pages?.flatMap((page) => page.rows) ?? [],
+    [data]
+  ) as OutOfOfficeEntry[];
+  const totalFetched = flatData.length;
+
+  const columns: ColumnDef<OutOfOfficeEntry>[] = [];
+  if (selectedTab === OutOfOfficeTab.TEAM) {
+    columns.push({
+      id: "member",
+      header: `Member`,
+      cell: ({ row }) => {
+        if (!row.original || !row.original.user || isPending || isFetching) {
+          return <SkeletonText className="h-8 w-full" />;
+        }
+        const { avatarUrl, username, email } = row.original.user;
+        return (
+          <div className="flex items-center gap-2">
+            <Avatar
+              size="sm"
+              alt={username || email}
+              imageSrc={getUserAvatarUrl({
+                avatarUrl,
+              })}
+            />
+            <div className="">
+              <div
+                data-testid={`member-${username}-username`}
+                className="text-emphasis text-sm font-medium leading-none">
+                {username || "No username"}
```

Comment:
Refer to our team member view to see how we display the name instead of the username

Reply:
updated